plugins/browser-plugin-ga-cookies/src/index.ts

```diff
@@ -28,27 +28,67 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { SelfDescribingJson } from '@snowplow/tracker-core';
+import { LOG, SelfDescribingJson } from '@snowplow/tracker-core';
 import { BrowserPlugin, cookie } from '@snowplow/browser-tracker-core';
-import { Cookies } from './contexts';
+import { GA4Cookies, UACookies } from './types';
+import { GOOGLE_ANALYTICS_4_COOKIES_SCHEMA, UNIVERSAL_ANALYTICS_COOKIES_SCHEMA } from './schemata';
+
+interface GACookiesPluginOptions {
+  ua?: boolean;
+  ga4: boolean;
+  ga4ContainerId: string;
+}
+
+const defaultPluginOptions = {
+  ua: true,
+  ga4: false,
+  ga4ContainerId: '',
+};
 
 /**
  * Captures the GA cookies on a page and sends as context on each event
+ * @param pluginOptions.ua - Send Universal Analytics specific cookie values.
+ * @param pluginOptions.ga4 - Send Google Analytics 4 specific cookie values.
+ * @param pluginOptions.ga4ContainerId - Container id to search the Google Analytics 4 session cookie. The cookie has the form of _ga_<container-id> where <container-id> is the data stream container id.
  */
-export function GaCookiesPlugin(): BrowserPlugin {
+export function GaCookiesPlugin(pluginOptions: GACookiesPluginOptions = defaultPluginOptions): BrowserPlugin {
   return {
     contexts: () => {
-      const gaCookieData: SelfDescribingJson<Cookies> = {
-        schema: 'iglu:com.google.analytics/cookies/jsonschema/1-0-0',
-        data: {},
-      };
-      ['__utma', '__utmb', '__utmc', '__utmv', '__utmz', '_ga'].forEach(function (cookieType) {
-        var value = cookie(cookieType);
-        if (value) {
-          gaCookieData.data[cookieType] = value;
+      const contexts: SelfDescribingJson<Record<string, unknown>>[] = [];
+      const { ga4, ga4ContainerId, ua } = { ...defaultPluginOptions, ...pluginOptions };
+
+      if (ua) {
+        const uaCookiesContext: SelfDescribingJson<UACookies> = {
+          schema: UNIVERSAL_ANALYTICS_COOKIES_SCHEMA,
+          data: {},
+        };
+        ['__utma', '__utmb', '__utmc', '__utmv', '__utmz', '_ga'].forEach(function (cookieType) {
+          var value = cookie(cookieType);
+          if (value) {
+            uaCookiesContext.data[cookieType] = value;
+          }
+        });
+        contexts.push(uaCookiesContext);
+      }
+
+      if (ga4) {
+        if (!ga4ContainerId) {
+          LOG.warn('"ga4ContainerId" option not provided in browser-plugin-ga-cookies.');
```

Comment:
Should we just grab everything in `_ga_.+` in this case instead?

Reply:
I did not add this feature for now, if you think it might be a useful option, feel free to reopen.